[Geant4] Update vecgeom and clhep version, get geant4 from it's github

cmsbuild opened the pull request:
backport of #6486

please test
following https://github.com/cms-externals/geant4/issues

cmsbuild commented:
A new Pull Request was created by  @cmsbuild  for branch IB/CMSSW_11_3_X/geant4.

@cmsbuild, @smuzaffar, @mrodozov can you please review it and eventually sign? Thanks.
cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

smuzaffar commented:
please test

mrodozov commented:
please test with https://github.com/cms-sw/cmssw/pull/32420
well it also needs an IB that has the cmssw change , or the cmssw PR

cmsbuild commented:
-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-f3a3d6/11738/summary.html
**CMSSW**: CMSSW_11_3_GEANT4_X_2020-12-15-2300/slc7_amd64_gcc900


## Build

I found compilation error when building: 

<pre>>> Checking EDM Class Version for src/DataFormats/GsfTrackReco/src/classes_def.xml in libDataFormatsGsfTrackReco.so
@@@@ ----> OK  EDM Class Transients 
>> Checking EDM Class Version for src/DataFormats/HGCalReco/src/classes_def.xml in libDataFormatsHGCalReco.so
@@@@ ----> OK  EDM Class Transients 
>> Checking EDM Class Version for src/DataFormats/HLTReco/src/classes_def.xml in libDataFormatsHLTReco.so
error: class 'CLHEP::Hep3Vector' has a different checksum for ClassVersion 10. Increment ClassVersion to 11 and assign it to checksum 3438801427
Suggestion: You can run 'scram build updateclassversion' to generate src/DataFormats/CLHEP/src/classes_def.xml.generated with updated ClassVersion
gmake: *** [tmp/slc7_amd64_gcc900/edm_checks/libDataFormatsCLHEP.so] Error 1
@@@@ ----> OK  EDM Class Transients 
>> Checking EDM Class Version for src/DataFormats/HcalIsolatedTrack/src/classes_def.xml in libDataFormatsHcalIsolatedTrack.so
Error in <TCling::RegisterModule>: Dictionary trigger function for DataFormatsCLHEP_xr not found
</pre>

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-f3a3d6/11745/summary.html
**CMSSW**: CMSSW_11_3_GEANT4_X_2020-12-15-2300/slc7_amd64_gcc900

Found compilation warnings

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 47493 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2747284
 - DQMHistoTests: Total failures: 299838
 - DQMHistoTests: Total nulls: 552
 - DQMHistoTests: Total successes: 2446872
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -1.65 KiB( 35 files compared)
 - DQMHistoSizes: changed ( 10224.0 ): -0.488 KiB       SiStrip/MechanicalView
 - DQMHistoSizes: changed ( 250202.181 ): -0.404 KiB       SiStrip/MechanicalView
 - DQMHistoSizes: changed ( 25202.0 ): 0.171 KiB        SiStrip/MechanicalView
 - DQMHistoSizes: changed ( 7.3 ): -3.326 KiB       SiStrip/MechanicalView
 - DQMHistoSizes: changed ( 8.0 ): 2.397 KiB        SiStrip/MechanicalView
 - Checked 153 log files, 37 edm output root files, 36 DQM output files

civanch commented:
Modified SIM history is expected.

mrodozov commented:
+externals

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next IB/CMSSW_11_3_X/geant4 IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)
